MapboxNavigation/RouteManeuverViewController.swift

```diff
@@ -13,10 +13,19 @@ class RouteManeuverViewController: UIViewController {
     let routeStepFormatter = RouteStepFormatter()
     let visualInstructionFormatter = VisualInstructionFormatter()
     
-    var step: RouteStep? {
+    var upcomingStep: RouteStep? {
         didSet {
             if isViewLoaded {
-                roadCode = step?.codes?.first ?? step?.destinationCodes?.first ?? step?.destinations?.first
+                roadCode = upcomingStep?.codes?.first ?? upcomingStep?.destinationCodes?.first ?? upcomingStep?.destinations?.first
+                updateStreetNameForStep()
+            }
+        }
+    }
+    
+    var currentStep: RouteStep? {
```

Comment:
Setting the `currentStep` should automatically update `upcomingStep` too; otherwise, weird things could happen if we let the two properties get out of sync. Instead of maintaining parallel state, can we extend Route with a method that gets the step after a given step? Would that be efficient enough for our needs here?

Reply:
We don't have access to route or routeprogress here. Will look around.

Reply:
@1ec5 changed this around a bit here https://github.com/mapbox/mapbox-navigation-ios/pull/767/commits/774898dd53431370ea60f3b03fa2c2a14298d534